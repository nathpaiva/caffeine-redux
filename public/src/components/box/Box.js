import React from 'react';
import { Link } from 'react-router-dom';

import Container from '../container/Container';
import Input from '../input/Input';
import Button from '../button/Button';
import Title from '../title/Title';

import style from '../../css/inline';

const Box = ({ link, title, text, inputs, button, to }) => (
  <section style={style.boxform}>
    <ul style={style.box}>
      {!!link && (
        <div style={style.containterLink}>
          {/* card link */}
          {link.map((item, i) => {
            if (item.to) {
              return (
                <Link key={i} style={style.editLink} to={item.to}>
                  {item.text}
                </Link>
              );
            } else {
              return (
                <a key={i} style={style.editLink} onClick={item.action}>
                  {item.text}
                </a>
              );
            }
          })}
        </div>
      )}
      {!!title && (
        <Container>
          <Title title={title} />
        </Container>
      )}
      {!!text && (
        <Container>
          <p>{text}</p>
        </Container>
      )}
      {!!inputs &&
        inputs.map((input, i) => {
          return (
            <Container key={i}>
              <Input
                type={input.type}
                id={input.id}
                text={input.text}
                inputRef={input.inputRef}
                label={input.label}
                disabled={input.disabled}
                defaultValue={input.defaultValue}
              />
            </Container>
          );
        })}
      {!!button && (
        <Container>
          {(!!to && (
            <Link to={to}>
              <Button name={button} />
            </Link>
          )) || <Button name={button} />}
        </Container>
      )}
    </ul>
  </section>
);

export default Box;
