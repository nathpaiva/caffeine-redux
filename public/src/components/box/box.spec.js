'use strict';

import React from 'react';
import renderer from 'react-test-renderer';

import Box from './Box';

describe('#Test Box Component', () => {
  it('#Should Box with success', () => {
    const tree = renderer
      .create(
        <Box
          title='Register'
          text='First time using of caffeine.'
          inputs={[
            {
              type: 'email',
              id: 'email',
              text: 'email',
              inputRef: (input) => {
                var email = input;
                return email;
              },
            },
            {
              type: 'text',
              id: 'login',
              text: 'login',
              inputRef: (input) => {
                var login = input;
                return login;
              },
            },
            {
              type: 'password',
              id: 'pass',
              text: 'password',
              inputRef: (input) => {
                var password = input;
                return password;
              },
            },
          ]}
          button='Create'
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
