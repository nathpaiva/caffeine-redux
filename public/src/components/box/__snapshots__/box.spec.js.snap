// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#Test Box Component #Should Box with success 1`] = `
<section
  style={
    Object {
      "backgroundColor": "#fff",
      "border": "1px solid rgb(90, 74, 105)",
      "borderRadius": "2px",
      "display": "flex",
      "margin": "10px auto auto",
      "maxWidth": "500px",
      "padding": "10px 0",
      "width": "100%",
    }
  }
>
  <ul
    style={
      Object {
        "padding": "0 10px",
        "position": "relative",
        "width": "100%",
      }
    }
  >
    <li
      style={
        Object {
          "fontSize": "12px",
          "margin": "5px 0",
        }
      }
    >
      <h2
        style={
          Object {
            "color": "#414141",
            "fontFamily": "\\"Roboto\\", sans-serif",
            "fontSize": "20px",
            "textAlign": "center",
          }
        }
      >
        Register
      </h2>
    </li>
    <li
      style={
        Object {
          "fontSize": "12px",
          "margin": "5px 0",
        }
      }
    >
      <p>
        First time using of caffeine.
      </p>
    </li>
    <li
      style={
        Object {
          "fontSize": "12px",
          "margin": "5px 0",
        }
      }
    >
      <div>
        
        <input
          defaultValue=""
          disabled={false}
          id="email"
          name="email"
          placeholder="email"
          style={
            Object {
              "border": "1px solid #CCCCCC",
              "borderRadius": "2px",
              "height": "32px",
              "margin": "2px 0",
              "padding": "0 5px",
              "width": "100%",
            }
          }
          type="email"
        />
      </div>
    </li>
    <li
      style={
        Object {
          "fontSize": "12px",
          "margin": "5px 0",
        }
      }
    >
      <div>
        
        <input
          defaultValue=""
          disabled={false}
          id="login"
          name="login"
          placeholder="login"
          style={
            Object {
              "border": "1px solid #CCCCCC",
              "borderRadius": "2px",
              "height": "32px",
              "margin": "2px 0",
              "padding": "0 5px",
              "width": "100%",
            }
          }
          type="text"
        />
      </div>
    </li>
    <li
      style={
        Object {
          "fontSize": "12px",
          "margin": "5px 0",
        }
      }
    >
      <div>
        
        <input
          defaultValue=""
          disabled={false}
          id="pass"
          name="pass"
          placeholder="password"
          style={
            Object {
              "border": "1px solid #CCCCCC",
              "borderRadius": "2px",
              "height": "32px",
              "margin": "2px 0",
              "padding": "0 5px",
              "width": "100%",
            }
          }
          type="password"
        />
      </div>
    </li>
    <li
      style={
        Object {
          "fontSize": "12px",
          "margin": "5px 0",
        }
      }
    >
      <button
        style={
          Object {
            "backgroundColor": "#5A4A69",
            "border": "none",
            "borderRadius": "2px",
            "color": "#FFFFFF",
            "cursor": "pointer",
            "fontSize": "12px",
            "fontWeight": "bold",
            "padding": "10px",
            "textAlign": "center",
            "textTransform": "uppercase",
            "width": "100%",
          }
        }
      >
        Create
      </button>
    </li>
  </ul>
</section>
`;
